### PR TITLE
Reduce duplication in the unit tests

### PR DESCRIPTION
**Pull request by clumens**
This can be left until after other unit test PRs get merged.  I'll have to update it, but I don't want to keep blocking those other PRs with my changes.

### CODE REVIEW

File: lib/common/tests/acl/xml_acl_denied_test.c [hunk 1]
```diff
@@ -56,14 +56,6 @@ is_xml_acl_denied_with_node(void **state)
...
-{
-    const struct CMUnitTest tests[] = {
-        cmocka_unit_test(is_xml_acl_denied_without_node),
-        cmocka_unit_test(is_xml_acl_denied_with_node),
-    };
-
-    cmocka_set_message_output(CM_OUTPUT_TAP);
-    return cmocka_run_group_tests(tests, NULL, NULL);
-}
+PCMK__UNIT_TEST(NULL, NULL, \
```
Comment (kgaillot) [inline lib/common/tests/acl/xml_acl_denied_test.c:59]: don't need backslash

**Review by kgaillot — COMMENTED** [review]
Yeah this will really help simplify things

File: lib/common/tests/acl/xml_acl_denied_test.c (hunk 1 above)
Comment (clumens) [reply lib/common/tests/acl/xml_acl_denied_test.c:59]: Fixed.

**Comment by clumens** [comment]
Hopefully we're just about at a point where we don't need to do anything to every test file.  We're starting to have so many that it's obnoxious to do.

File: doc/sphinx/Pacemaker_Development/helpers.rst [hunk 2]
```diff
@@ -224,17 +224,13 @@ is no ``lib/common/tests/acls`` directory.
...
+  are included in the ``Makefile.am``.  These files contain all the flags necessary
+  for most unit tests.  If necessary, individual settings can be overridden like so:
 
   .. code-block:: none
 
-     LDADD = $(top_builddir)/lib/common/libcrmcommon_test.la \
-             -lcmocka
-     AM_CFLAGS = -DPCMK__UNIT_TESTING
-     AM_LDFLAGS = $(LDFLAGS_WRAP)
+     LDADD += $(top_builddir)/lib/pengine/libpe_status_test.la
```
Comment (nrwahl2) [inline doc/sphinx/Pacemaker_Development/helpers.rst:233]: There might be clearer examples of overriding individual settings than appending a `.la` file

File: doc/sphinx/Pacemaker_Development/helpers.rst [hunk 3]
```diff
@@ -467,6 +467,46 @@ At this point, you need to determine whether your test case is incorrect or
 whether the code being tested is incorrect.  Fix whichever is wrong and continue.
 
 
+Code Coverage
```
Comment (nrwahl2) [inline doc/sphinx/Pacemaker_Development/helpers.rst:470]: Thanks :)

**Comment by nrwahl2** [comment]
Big fan

File: doc/sphinx/Pacemaker_Development/helpers.rst (hunk 2 above)
Comment (clumens) [reply doc/sphinx/Pacemaker_Development/helpers.rst:233]: I've updated this with the only other thing we're overriding at the moment, which is AM_CPPFLAGS in one spot.

**Comment by clumens** [comment]
I've rebased this on master.  It's good to go now.

**Pull request merged**